libs/feature/editor/src/lib/components/record-form/form-field/form-field-constraints-shortcuts/form-field-constraints-shortcuts.component.ts

```diff
@@ -137,60 +145,42 @@ export class FormFieldConstraintsShortcutsComponent implements OnInit {
     this.editorFacade.setFieldVisibility({ model: 'otherConstraints' }, false)
   }
 
-  showSpecificConstraintsTextAreas(constraintSection: ConstraintChoice) {
-    this.editorFacade.setFieldVisibility({ model: constraintSection }, true)
-  }
-
-  constructor(private editorFacade: EditorFacade) {}
-
-  onToggleChange(toggleName: string) {
-    if (toggleName === 'toggleApplicableConstraint') {
-      this.toggleApplicableConstraint = !this.toggleApplicableConstraint
-
-      if (this.toggleApplicableConstraint) {
-        this.toggleKnownConstraint = false // if toggleApplicableConstraint is turned on, toggleKnownConstraint must be off
-
-        this.editorFacade.updateRecordField('legalConstraints', [
-          { text: NOT_APPLICABLE_CONSTRAINT },
-        ])
-        this.hideAllConstraintSections()
-      } else {
-        // if only toggle is turned off, remove the constraint
-        this.editorFacade.updateRecordField('legalConstraints', []) //remove all legal constraints
-        // TODO: show the constraint sections that are already present in the record
-      }
-    } else if (toggleName === 'toggleKnownConstraint') {
-      this.toggleKnownConstraint = !this.toggleKnownConstraint
-
-      if (this.toggleKnownConstraint) {
-        this.toggleApplicableConstraint = false // if toggleKnownConstraint is turned on, toggleApplicableConstraint must be off
-
-        this.editorFacade.updateRecordField('legalConstraints', [
-          { text: NOT_KNOWN_CONSTRAINT },
-        ])
-        this.hideAllConstraintSections()
-      } else {
-        // if only toggle is turned off, remove the constraint
-        this.editorFacade.updateRecordField('legalConstraints', []) //remove all legal constraints
-        // TODO: show the constraint sections that are already present in the record
-      }
+  onToggleChange(
+    toggleName: 'noApplicableConstraint' | 'noKnownConstraint',
+    value: boolean
+  ) {
+    if (value) {
+      const presetConstraint =
+        toggleName === 'noApplicableConstraint'
+          ? NOT_APPLICABLE_CONSTRAINT
+          : NOT_KNOWN_CONSTRAINT
+      this.editorFacade.updateRecordField('legalConstraints', [
+        presetConstraint,
+      ])
+      this.hideAllConstraintSections()
+    } else {
+      // if the toggle is turned off, remove the constraint
+      this.editorFacade.updateRecordField('legalConstraints', []) //remove all legal constraints
```

Comment:
Let's assume we edit a record where there are some legal constraints already present.
Then we toggle either of the toggle possibilites and untoggle the same one again. After that all legal constraints that were previously in the record will be gone, right? Is that ok for the user?

Reply:
you're right, that's probably too radical; we should only remove constraints that match the "no applicable constraints" or "no known constraints" criteria